Lib/test/test_inspect/test_inspect.py

```diff
@@ -1816,49 +1816,103 @@ def test_typing_replacement(self):
 class TestIsMethodDescriptor(unittest.TestCase):
 
     def test_custom_descriptors(self):
+        # Custom method descriptors (non-data descriptors):
+
         class MethodDescriptor:
+            """with __get__ and no __set__/__delete__"""
             def __get__(self, *_): pass
+        class MethodDescriptorWithGetNone:
+            """with __get__=None and no __set__/__delete__"""
+            __get__ = None
         class MethodDescriptorSub(MethodDescriptor):
-            pass
-        class DataDescriptorWithNoGet:
+            """with __get__ (inherited) and no __set__/__delete__"""
+
+        for cls in [
+            MethodDescriptor,
+            MethodDescriptorWithGetNone,
+            MethodDescriptorSub,
+        ]:
+            with self.subTest(f"a descriptor {cls.__doc__}", cls=cls):
+                self.assertTrue(inspect.ismethoddescriptor(cls()))
```

Comment:
Method descriptors and data descriptors are mutually exclusive. For every case where `ismethoddescriptor()` returns True we can test that `isdatadescriptor()` returns False and vice verse. Of course, there are objects for which both predicates return False.

Reply:
Generally, both those functions – `ismethoddescriptor()` and `isdatadescriptor()` – are tested in similar ways, but separately: each in a dedicated test class (respectively: `TestIsMethodDescriptor` and `TestIsDataDescriptor`).

Do you suggest that those classes should be merged into one class? That seems reasonable from the DRY point of view. On the other hand, I am afraid if it wouldn't result in tests whose essence would be harder to grasp (because of being less focused).